### PR TITLE
Endpoint for agent autocomplete

### PR DESCRIPTION
**Pull request by philipperolet**
As discussed in item #1345

### CODE REVIEW

File: front/pages/api/w/[wId]/assistants/autocomplete.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      api_error: {
+        type: "invalid_request_error",
+        message: "The agentPrefix parameter is required.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const assistants = await getAgentConfigurations(auth);
```
Comment (spolu) [inline front/pages/api/w/[wId]/assistants/autocomplete.ts:54]: Missing prefix?

File: front/lib/api/assistant/configuration.ts [hunk 2]
```diff
@@ -171,6 +173,13 @@ export async function getAgentConfigurations(
   const rawAgents = await AgentConfiguration.findAll({
     where: {
       workspaceId: owner.id,
+      ...(agentPrefix
+        ? {
+            name: {
+              [Op.iLike]: `${agentPrefix}%`,
```
Comment (PopDaph) [inline front/lib/api/assistant/configuration.ts:179]: You might also want to filter in `getGlobalAgents` 🙏🏻 
Workspace agents are stored in database, and Global Agents defined by Dust are stored directly on the code, as they are built and retrieved from the user's workspace.

File: front/pages/api/w/[wId]/assistants/autocomplete.ts (hunk 1 above)
Comment (philipperolet) [reply front/pages/api/w/[wId]/assistants/autocomplete.ts:54]: :fearful: :sweat_smile:

**Review by PopDaph — APPROVED** [review]

**Review by spolu — APPROVED** [review]

**Comment by spolu** [comment]
Note for future self, this is likely what we'll want to do:
```
CREATE INDEX idx_lower_colname ON tablename (lower(colname));
SELECT * FROM tablename WHERE lower(colname) LIKE lower('%pattern%');
```

**Pull request merged**